File: C++/Wolf.cpp
#include "CONSTANTS.h"

#include "Wolf.h"

Wolf::Wolf(World &world, const Point &position) : Animal(world, position, WOLF_SYMBOL, WOLF_STRENGTH, WOLF_INITIATIVE, WOLF_COLOR, WOLF_NAME) {}

Organism* Wolf::Clone(const Point &position) {
  return new Wolf(this->world, position);
}

File: Java/Wolf.java
public class Wolf extends Animal {
  public Wolf(World world, Point position) {
    super(world, position, Constants.WOLF_SYMBOL, Constants.WOLF_STRENGTH, Constants.WOLF_INITIATIVE, Constants.WOLF_COLOR, Constants.WOLF_NAME, Constants.ANIMAL_RANGE);
  }

  @Override
  public final Organism Clone(Point position) {
    return new Wolf(this.world, position);
  }
}
